channel.go

```diff
@@ -217,6 +217,35 @@ func (c *ChannelService) UpdateChannelMemberNotifications(channelID, userID stri
 	return channelMember, normalizeAppErr(appErr)
 }
 
+// CreateSidebarCategory creates a new sidebar category for a set of channels.
+//
+// Minimum server version: 5.37
+func (c *ChannelService) CreateSidebarCategory(
+	userID, teamID string, newCategory *model.SidebarCategoryWithChannels) (*model.SidebarCategoryWithChannels, error) {
```

Comment:
The general design guideline for the API has been to not return an point to the object, which was created and instead overwrite the passed pointer as in https://github.com/mattermost/mattermost-plugin-api/blob/e6371ea3fdddd63a2d436be72625424accbd1241/user.go#L93-L99. Is the a reason this PR deviates from that pattern?

Reply:
I did not know about this convention, but I'm all in for consistency. @madhavhugar, do you mind following the pattern suggested by @hanzei?

Reply:
@hanzei @agarciamontoro the pattern makes sense to reduce memory footprint, I've made the changes